pkg/providers/tinkerbell/validate_test.go

```diff
@@ -0,0 +1,31 @@
+package tinkerbell
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+)
+
+func TestPermuatations(t *testing.T) {
+	tests := []struct {
+		initial    string
+		separators []string
+		want       string
+	}{
+		{"125", []string{"-", "_", ""}, "125"},
+		{"1.26", []string{"-", "_", ""}, "1.26, 1-26, 1_26 or 126"},
+		{"1.27", []string{"."}, "1.27"},
+		{"1.28", []string{".", "."}, "1.28"},
+		{"1.29", []string{"-", "-", ""}, "1.29, 1-29 or 129"},
+		{"1.29.1", []string{"-", "_", "_", ""}, "1.29.1, 1-29-1, 1_29_1 or 1291"},
```

Comment:
we don't support patch version on our spec, so we don't have to be covering this. hence a simpler approach might work?

Reply:
what versions we support is business level logic. Coupling this with application logic will cause us future maintainability issues in the future. I don't recommend we couple them.